analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,288 @@
+  - **`CD4_Tcells`**
+  - **`CD8_Tcells`**
+  - `Monocytes`
+  - `Neutrophils`
+  - **`NKcells`**
+  - `otherCells`
+          
+`EPIC` was run on bulk samples from five different projects with the high-level diagnoses shown in the table below.
+Based on the biology of these groups of tumors, we plan to analyze each with the reference indicated in the table.
+
+
+| Project     | High-level diagnosis | `EPIC` Reference |
+|-------------|----------------------|------------------|
+| SCPCP000001 | High-grade glioma    | `TRef`           |   
+| SCPCP000002 | Low-grade glioma     | `TRef`           |   
+| SCPCP000006 | Wilms tumor          | `BRef`           |   
+| SCPCP000009 | CNS tumors           | `TRef`           |   
+| SCPCP000017 | Osteosarcoma         | `BRef`           |   
+
+
+## Setup
+
+```{r setup, message = FALSE}
+renv::load()
+
+library(ggplot2)
+theme_set(theme_bw())
+```
+
+
+### Paths
+
+```{r paths}
+results_dir <- here::here("analysis", "bulk-deconvolution", "results")
+
+epic_files <- list.files(
+  path = results_dir,
+  full.names = TRUE,
+  pattern = "*-epic.tsv$"
+)
+project_ids <- stringr::str_split_i(basename(epic_files), pattern = "-", i = 1)
+epic_files <- setNames(epic_files, project_ids)
+```
+
+### Prepare input data
+
+First, we'll read in the `EPIC` results and subset to only the reference of interest for each. 
+
+```{r}
+tref_projects <- c("SCPCP000001", "SCPCP000002", "SCPCP000009")
+bref_projects <- c("SCPCP000006", "SCPCP000017")
+
+# Read in epic result TSVs and combine into single data frame
+epic_raw_df <- epic_files |>
+  purrr::map(\(file) readr::read_tsv(file, show_col_types = FALSE)) |>
+  purrr::list_rbind(names_to = "project_id") 
+
+# Filter based on projects and references
+epic_tref <- epic_raw_df |>
+  dplyr::filter(
+    project_id %in% tref_projects,
+    reference == "TRef"
+  )
+ 
+epic_bref <- epic_raw_df |>
+  dplyr::filter(
+    project_id %in% bref_projects,
+    reference == "BRef"
+  )
+
+# Create final epic_df with only references of interest for each project
+epic_df <- dplyr::bind_rows(
+  epic_tref, 
+  epic_bref
+) |>
+  # add column to support interpreting figure panels with projects, diagnosis, and EPIC reference
+ dplyr::mutate(
+   project = dplyr::case_match(
+      project_id, 
+      "SCPCP000001" ~ "01: High-grade glioma (TRef)", 
+      "SCPCP000002" ~ "02: Low-grade glioma (TRef)", 
+      "SCPCP000006" ~ "06: Wilms tumor (BRef)", 
+      "SCPCP000009" ~ "09: CNS tumors (TRef)", 
+      "SCPCP000017" ~ "17: Osteosarcoma (BRef)", 
+    ), 
+   # arrange in order of reference, not project
+   project = forcats::fct_relevel(project, "09: CNS tumors (TRef)", after = 2)
+  ) |>
+  dplyr::select(project, sample_id, epic_celltype, fraction)
+
+head(epic_df)
+```
+
+
+
+## Visualization
+
+### Proportions of cell types
+
+Considering all categories of cells, what proportions of cell types are represented across samples?
+
+
+```{r fig.height=8, fig.width=12}
```

Comment:
Thank you for making these panels bigger! But I actually think we want to show the results for both references for each reference here? I think that drives home the point of why you are choosing one ref over the other in many cases.

Reply:
After going through the rest of the notebook, I would show this plot for both references for all projects and then include reasoning on why you are picking each reference for each project. And then for the rest of the notebook you could just use the one you pick.